// chia-agent/api/ws/wallet/index.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.on_message_from_wallet = exports.on_coin_added_of_wallet = exports.coin_added_command_of_wallet = exports.on_sync_changed_of_wallet = exports.sync_changed_command_of_wallet = exports.on_state_changed_of_wallet = exports.state_changed_command_of_wallet = exports.on_get_connections = exports.get_connections_command = exports.chia_wallet_service = void 0;
const types_1 = require("../../types");
exports.chia_wallet_service = "chia_wallet";
exports.get_connections_command = "get_connections";
function on_get_connections(daemon, callback) {
    return __awaiter(this, void 0, void 0, function* () {
        yield daemon.subscribe(types_1.wallet_ui_service);
        const messageListener = (e) => {
            if (e.origin === exports.chia_wallet_service && e.command === exports.get_connections_command) {
                callback(e);
            }
        };
        return daemon.addMessageListener(exports.chia_wallet_service, messageListener);
    });
}
exports.on_get_connections = on_get_connections;
exports.state_changed_command_of_wallet = "state_changed";
function on_state_changed_of_wallet(daemon, callback) {
    return __awaiter(this, void 0, void 0, function* () {
        yield daemon.subscribe(types_1.wallet_ui_service);
        const messageListener = (e) => {
            if (e.origin === exports.chia_wallet_service && e.command === exports.state_changed_command_of_wallet) {
                callback(e);
            }
        };
        return daemon.addMessageListener(exports.chia_wallet_service, messageListener);
    });
}
exports.on_state_changed_of_wallet = on_state_changed_of_wallet;
// "sync_changed" and "coin_added" below are for metrics services.
exports.sync_changed_command_of_wallet = "sync_changed";
function on_sync_changed_of_wallet(daemon, callback) {
    return __awaiter(this, void 0, void 0, function* () {
        yield daemon.subscribe(types_1.metrics_service);
        const messageListener = (e) => {
            if (e.origin === exports.chia_wallet_service && e.command === exports.sync_changed_command_of_wallet) {
                callback(e);
            }
        };
        return daemon.addMessageListener(exports.chia_wallet_service, messageListener);
    });
}
exports.on_sync_changed_of_wallet = on_sync_changed_of_wallet;
exports.coin_added_command_of_wallet = "coin_added";
function on_coin_added_of_wallet(daemon, callback) {
    return __awaiter(this, void 0, void 0, function* () {
        yield daemon.subscribe(types_1.metrics_service);
        const messageListener = (e) => {
            if (e.origin === exports.chia_wallet_service && e.command === exports.coin_added_command_of_wallet) {
                callback(e);
            }
        };
        return daemon.addMessageListener(exports.chia_wallet_service, messageListener);
    });
}
exports.on_coin_added_of_wallet = on_coin_added_of_wallet;
function on_message_from_wallet(daemon, callback) {
    return __awaiter(this, void 0, void 0, function* () {
        yield daemon.subscribe(types_1.wallet_ui_service);
        const messageListener = (e) => {
            if (e.origin === exports.chia_wallet_service) {
                callback(e);
            }
        };
        return daemon.addMessageListener(exports.chia_wallet_service, messageListener);
    });
}
exports.on_message_from_wallet = on_message_from_wallet;
