// chia-agent/api/ws/timelord/index.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.on_new_peak = exports.new_peak_command = exports.on_skipping_peak = exports.skipping_peak_command = exports.on_new_compact_proof = exports.new_compact_proof_command = exports.on_finished_pot = exports.finished_pot_command = exports.chia_timelord_service = void 0;
const types_1 = require("../../types");
exports.chia_timelord_service = "chia_timelord";
exports.finished_pot_command = "finished_pot";
function on_finished_pot(daemon, callback) {
    return __awaiter(this, void 0, void 0, function* () {
        yield daemon.subscribe(types_1.metrics_service);
        const messageListener = (e) => {
            if (e.origin === exports.chia_timelord_service && e.command === exports.finished_pot_command) {
                callback(e);
            }
        };
        return daemon.addMessageListener(exports.chia_timelord_service, messageListener);
    });
}
exports.on_finished_pot = on_finished_pot;
exports.new_compact_proof_command = "new_compact_proof";
function on_new_compact_proof(daemon, callback) {
    return __awaiter(this, void 0, void 0, function* () {
        yield daemon.subscribe(types_1.metrics_service);
        const messageListener = (e) => {
            if (e.origin === exports.chia_timelord_service && e.command === exports.new_compact_proof_command) {
                callback(e);
            }
        };
        return daemon.addMessageListener(exports.chia_timelord_service, messageListener);
    });
}
exports.on_new_compact_proof = on_new_compact_proof;
exports.skipping_peak_command = "skipping_peak";
function on_skipping_peak(daemon, callback) {
    return __awaiter(this, void 0, void 0, function* () {
        yield daemon.subscribe(types_1.metrics_service);
        const messageListener = (e) => {
            if (e.origin === exports.chia_timelord_service && e.command === exports.skipping_peak_command) {
                callback(e);
            }
        };
        return daemon.addMessageListener(exports.chia_timelord_service, messageListener);
    });
}
exports.on_skipping_peak = on_skipping_peak;
exports.new_peak_command = "new_peak";
function on_new_peak(daemon, callback) {
    return __awaiter(this, void 0, void 0, function* () {
        yield daemon.subscribe(types_1.metrics_service);
        const messageListener = (e) => {
            if (e.origin === exports.chia_timelord_service && e.command === exports.new_peak_command) {
                callback(e);
            }
        };
        return daemon.addMessageListener(exports.chia_timelord_service, messageListener);
    });
}
exports.on_new_peak = on_new_peak;
